example_list_events_test.go

```diff
@@ -0,0 +1,81 @@
+package paddle_test
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+
+	paddle "github.com/PaddleHQ/paddle-go-sdk"
+)
+
+// Demonstrates how to fetch a list of events and iterate over the provided results.
+func Example_listEvents() {
+	// Create a mock HTTP server for this example - skip over this bit!
+	s := mockServerForExample(mockServerResponse{stub: &stub{paths: []stubPath{events}}})
+
+	// Create a new Paddle client.
+	client, err := paddle.New(
+		os.Getenv("PADDLE_API_KEY"),
+		paddle.WithBaseURL(s.URL), // Uses the mock server, you will not need this in your integration.
+	)
+
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
+
+	ctx := context.Background()
+
+	// Optionally set a transit ID on the context. This is useful to link your
+	// own request IDs to Paddle API requests.
+	ctx = paddle.ContextWithTransitID(ctx, "sdk-testing-request-1")
+
+	// Get a collection of events.
+	res, err := client.ListEvents(ctx, &paddle.ListEventsRequest{})
+
+	// Iterate the events.
+	err = res.Iter(ctx, func(e paddle.Event) (bool, error) {
+		switch v := e.(type) {
+		case *paddle.TransactionCompletedEvent:
+			// here v could be used as concrete type TransactionCompletedEvent
+			fmt.Println(v.EventID)
+			fmt.Println(v.Data.ID)
+		case *paddle.TransactionUpdatedEvent:
+			// here v could be used as concrete type TransactionUpdatedEvent
+			fmt.Println(v.EventID)
+			fmt.Println(v.Data.ID)
+		default:
```

Comment:
Thoughts about making a case for GenericEvent (I believe it should be this type of it's an event not handled by current SDK) and then having default as more of an unhandled case, log and error?

Reply:
I guess it depends of what we want to do with it, there might be cases where you only care about couple of events like we do in the example and the rest you just want to log it and nothing else.

Reply:
Having a `GenericEvent` (or `UnknownEvent`?) is less disruptive than erroring.

If a new event is subscribed to, we wouldn't want that to potentially break an integration running in production. If the integrators implementation isn't handling a default case (or `GenericEvent`) there's still a chance that subscribing to a new event causes an error on their side, but perhaps lower risk?

In any case we should perhaps document that integrators handle "unknown events"